File: frontend/inspect_package.js
import fs from 'fs';
import path from 'path';

const filePath = path.resolve('package.json');
try {
    const buffer = fs.readFileSync(filePath);
    console.log('Use Buffer');
    console.log('First 20 bytes:', buffer.subarray(0, 20));
    console.log('First 20 bytes hex:', buffer.subarray(0, 20).toString('hex'));
    
    const content = buffer.toString('utf8');
    console.log('Content starts with:', content.substring(0, 20));
    console.log('First char code:', content.charCodeAt(0));
    
    if (content.charCodeAt(0) === 0xFEFF) {
        console.log('BOM detected!');
    } else {
        console.log('No BOM detected.');
    }
    
    try {
        JSON.parse(content);
        console.log('JSON.parse successful');
    } catch (e) {
        console.log('JSON.parse failed:', e.message);
    }
} catch (e) {
    console.error('Error reading file:', e);
}

// Check for other config files
const potentialConfigs = [
    'postcss.config.js',
    '.postcssrc',
    '.postcssrc.js',
    '.postcssrc.json',
    'postcss.config.cjs'
];

console.log('Checking for potential PostCSS config files:');
potentialConfigs.forEach(config => {
    if (fs.existsSync(config)) {
        console.log(`Found: ${config}`);
    } else {
        console.log(`Not found: ${config}`);
    }
});

// Walk to find any file with 'postcss' in name
function walk(dir) {
    try {
        const files = fs.readdirSync(dir);
        files.forEach(file => {
            const filePath = path.join(dir, file);
            const stat = fs.statSync(filePath);
            if (stat.isDirectory()) {
                if (file !== 'node_modules' && file !== '.git') {
                    walk(filePath);
                }
            } else {
                if (file.toLowerCase().includes('postcss')) {
                    console.log(`Found in walk: ${filePath}`);
                }
            }
        });
    } catch (e) {
        console.error('Error reading dir ' + dir + ': ' + e.message);
    }
}
console.log('Walking directory to find *postcss* files:');
walk('.');

File: frontend/inspect_package.py
import os
import json

file_path = 'package.json'
try:
    with open(file_path, 'rb') as f:
        # Read start of file
        start_bytes = f.read(20)
        print(f"Start bytes: {start_bytes}")
        print(f"Start hex: {start_bytes.hex()}")
        
        # Read entire file and trying to decode
        f.seek(0)
        content = f.read()
        try:
            text = content.decode('utf-8')
            print("Decoded as utf-8 successfully")
            
            # Check for BOM
            if text.startswith('\ufeff'):
                print("BOM detected!")
            else:
                print("No BOM detected.")
                
            # Parse JSON
            try:
                json.loads(text)
                print("JSON parsing successful")
            except json.JSONDecodeError as e:
                print(f"JSON parsing failed: {e}")
                
        except UnicodeDecodeError:
            print("Failed to decode as utf-8")
            
except FileNotFoundError:
    print(f"{file_path} not found")

# Check for other config files
potential_configs = [
    'postcss.config.js',
    '.postcssrc',
    '.postcssrc.js',
    '.postcssrc.json',
    'postcss.config.cjs'
]

print("Checking for potential PostCSS config files:")
for config in potential_configs:
    if os.path.exists(config):
        print(f"Found: {config}")
    else:
        print(f"Not found: {config}")

# Walk to find any file with 'postcss' in name
print("Walking directory to find *postcss* files:")
for root, dirs, files in os.walk('.'):
    # skip node_modules
    if 'node_modules' in dirs:
        dirs.remove('node_modules')
    
    for file in files:
        if 'postcss' in file.lower():
            print(f"Found in walk: {os.path.join(root, file)}")
